apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -82,12 +92,38 @@ interface Context {
   isPatDeviceConnected: boolean;
   interpretation?: SheetOf<InterpretFileResult>;
   logger: BaseLogger;
+  paperHandlerDiagnosticElection?: ElectionDefinition;
+  diagnosticError?: DiagnosticError;
 }
 
 function assign(arg: Assigner<Context, any> | PropertyAssigner<Context, any>) {
   return xassign<Context, any>(arg);
 }
 
+class DiagnosticError extends Error {
+  constructor(
+    userFacingMessage: string,
```

Comment:
I think we've generally been trying to follow the pattern of having the frontend/UI define the error message copy and having backends just use a string error code to indicate which error happened.

Reply:
Agree that the backend defining error messages is hard to follow.

In this implementation, `DiagnosticError.userFacingMessage` eventually gets stored on the `DiagnosticRecord`. Since there's an existing pattern of messages being written to that table (from the frontend), I think there's a bigger refactor that would need to happen where

1. All writers of `DiagnosticRecord` pass an error code
2. We have a mapping of code to user-facing string
3. We reference those strings from report and diagnostic screen components

To avoid muddling the error landscape further I just replaced `DiagnosticError` with `Error`. Raw error messages will still be logged but no user-facing error will be saved to `DiagnosticRecord`.